hedera-node/src/main/java/com/hedera/services/state/exports/SignedStateBalancesExporter.java

```diff
@@ -246,7 +247,7 @@ private void addLegacyHeader(Writer writer, Instant at) throws IOException {
 
 	private void addRelease090Header(Writer writer, Instant at) throws IOException {
 		writer.write("# " + CURRENT_VERSION + LINE_SEPARATOR);
-		writer.write(String.format("# TimeStamp:%s%s", at, LINE_SEPARATOR));
+		writer.write("# TimeStamp:" + at + LINE_SEPARATOR);
```

Comment:
This is a good one. Also see the method exportBalancesFile. This is the real hotspot here and needs to be addressed.

Reply:
This is still inefficient to concat strings to then turn around and send it to a writer. Instead each sub-string should just be a separate `writer.write()`.
`writer.write("# TimeStamp:") + at + LINE_SEPARATOR);
```suggestion
		writer.write("# TimeStamp:");
		writer.write(at);
		writer.write(LINE_SEPARATOR);
```

Reply:
Wouldn't this approach increase the chances of IOExceptions?

Reply:
How so? You're still writing exactly the same amount of data. And the underlying writer is buffered so it's only writing to in memory objects until it's flushed.

Reply:
I see, thanks for the input. Made the changes.